modules/fundamental/src/error.rs

```diff
@@ -51,32 +51,53 @@ impl ResponseError for Error {
             Self::Purl(err) => {
                 HttpResponse::BadRequest().json(ErrorInformation::new("InvalidPurlSyntax", err))
             }
-            Self::Actix(err) => {
-                HttpResponse::InternalServerError().json(ErrorInformation::new("System Actix", err))
-            }
             Self::BadRequest { msg, status } => {
                 HttpResponse::build(*status).json(ErrorInformation::new("Bad request", msg))
             }
-            Self::Any(err) => HttpResponse::InternalServerError()
-                .json(ErrorInformation::new("System unknown", err)),
             Error::Ingestor(inner) => {
                 HttpResponse::BadRequest().json(ErrorInformation::new("Ingestor error", inner))
             }
-            Error::UnsupportedHashAlgorithm => HttpResponse::InternalServerError()
-                .json(ErrorInformation::new("Unsupported hash algorithm", "")),
-            Error::Storage(inner) => HttpResponse::InternalServerError()
-                .json(ErrorInformation::new("Unsupported hash algorithm", inner)),
-            Error::Database(err) => HttpResponse::InternalServerError()
-                .json(ErrorInformation::new("Database error", err)),
             Error::Query(err) => {
                 HttpResponse::BadRequest().json(ErrorInformation::new("Query error", err))
             }
             Error::IdKey(err) => HttpResponse::BadRequest().json(ErrorInformation::new("Key", err)),
             Error::StorageKey(err) => {
                 HttpResponse::BadRequest().json(ErrorInformation::new("Storage Key", err))
             }
-            Error::Data(msg) => HttpResponse::InternalServerError()
-                .json(ErrorInformation::new("Data-model corruption", msg)),
+
+            // The following match arms handle internal system errors that are not expected to occur.
+            // They are logged and a generic error response is returned to avoid leaking
+            // internal state to end users.
+            Self::Actix(err) => {
+                log::error!("System Actix: {}", err);
```

Comment:
I don't really like the pattern of adding a custom log message. IIRC we are using `thiserror` and so the error message should already be handled by the `Display` trait implementation. So a simple `log::error!("{err}")` should be sufficient. And given the fact that all match patterns would then have the same implementation, I think we can simplify this to just:

```rust
err => {
  log::error!("{err}");
  HttpResponse::InternalServerError()
    .json(ErrorInformation::new("Internal Server Error", ""))
}
```

Reply:
Sounds good.